sdk/storage/azure-storage-file-datalake/tests/testcase.py

```diff
@@ -71,37 +64,31 @@ def get_token(self, *args):
         return self.token
 
 
-class StorageTestCase(unittest.TestCase):
-
-    def setUp(self):
-        self.working_folder = os.path.dirname(__file__)
-
-        self.settings = settings
-        self.fake_settings = fake_settings
+class XMSRequestIDBody(RecordingProcessor):
+    """This process is used for Storage batch call only, to avoid the echo policy.
+    """
+    def process_response(self, response):
+        content_type = None
+        for key, value in response.get('headers', {}).items():
+            if key.lower() == 'content-type':
+                content_type = (value[0] if isinstance(value, list) else value).lower()
+                break
 
-        if settings is None:
-            self.test_mode = os.getenv('TEST_MODE') or TestMode.playback
-        else:
-            self.test_mode = self.settings.TEST_MODE.lower() or TestMode.playback
+        if content_type and 'multipart/mixed' in content_type:
+            response['body']['string'] = re.sub(b"x-ms-client-request-id: [a-f0-9-]+\r\n", b"", response['body']['string'])
 
-        if self.test_mode == TestMode.playback or (self.settings is None and self.test_mode.lower() == TestMode.run_live_no_record):
-            self.settings = self.fake_settings
+        return response
 
-        # example of qualified test name:
-        # test_mgmt_network.test_public_ip_addresses
-        _, filename = os.path.split(inspect.getsourcefile(type(self)))
-        name, _ = os.path.splitext(filename)
-        self.qualified_test_name = '{0}.{1}'.format(
-            name,
-            self._testMethodName,
-        )
 
+class StorageTestCase(AzureTestCase):
```

Comment:
StorageTestCase and AzureMgmtTestcase are both inheriting AzureTestcase. We don't need anything specific from either StorageTestCase/AzureMgmtTestcase, so there's no need to use them.
For example people who are using AzureMgmtTestcase is mainly because they need to create_mgmt_client to do some mgmt plane operation. For StorageTestCase we were using that to create storage account, while for datalake the storage account is created through ARM Template deployment so we don't need StorageTestCase either.
I know blob is using AzureMgmtTestcase probably when it was added we were using something?

Reply:
Ah ok that explains it - so blobs we have https://github.com/Azure/azure-sdk-for-python/blob/77feaf14471eba6642f5c7ae2f3f06981ff361d7/sdk/storage/azure-storage-blob/tests/_shared/testcase.py#L382 when we're creating the account, whereas in Datalake we're directly passing account name and key using the ARM template to deploy these resources (storage accounts)?